File: aspire-starter/aspire-starter.ApiService/Models/Department.cs
using System.ComponentModel.DataAnnotations;

namespace aspire_starter.ApiService.Models;

public class Department
{
    public int Id { get; set; }

    [Required, MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(500)]
    public string? Description { get; set; }

    [MaxLength(200)]
    public string? Location { get; set; }

    public int? HeadDoctorId { get; set; }
    public Doctor? HeadDoctor { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<Doctor> Doctors { get; set; } = [];
    public ICollection<Room> Rooms { get; set; } = [];
    public ICollection<Staff> StaffMembers { get; set; } = [];
}

File: aspire-starter/flutter_app/lib/models/department.dart
class Department {
  final int id;
  final String name;
  final String? description;
  final String? location;
  final int? headDoctorId;
  final DateTime createdAt;

  Department({
    this.id = 0,
    required this.name,
    this.description,
    this.location,
    this.headDoctorId,
    DateTime? createdAt,
  }) : createdAt = createdAt ?? DateTime.now();

  factory Department.fromJson(Map<String, dynamic> json) => Department(
    id: json['id'] as int? ?? 0,
    name: json['name'] as String? ?? '',
    description: json['description'] as String?,
    location: json['location'] as String?,
    headDoctorId: json['headDoctorId'] as int?,
    createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt'] as String) : DateTime.now(),
  );

  Map<String, dynamic> toJson() => {
    if (id > 0) 'id': id,
    'name': name,
    'description': description,
    'location': location,
    'headDoctorId': headDoctorId,
  };
}
